bazel/repository_locations.bzl

```diff
@@ -1052,6 +1052,21 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         release_date = "2022-10-12",
         cpe = "N/A",
     ),
+    intel_ittapi = dict(
```

Comment:
As per https://github.com/envoyproxy/envoy/blob/main/DEPENDENCY_POLICY.md#new-external-dependencies, please share details on extension checklist / scorecard. CC @envoyproxy/dependency-shepherds

Reply:
I have created an issue for the external dependency - not sure if tagging dependency-shepherds worked. https://github.com/envoyproxy/envoy/issues/28452

Reply:
See OSSF Scorecard numbers and suggestions at https://github.com/envoyproxy/envoy/issues/28452#issuecomment-1640110643